sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -32,6 +32,7 @@ import org.apache.spark.annotation.{Evolving, Stable}
 sealed abstract class Filter {
   /**
    * List of columns that are referenced by this filter.
+   * Note that, if a column contains `dots` in name, it will be quoted to avoid confusion.
```

Comment:
This one is actually a pretty breaking change. Not all implementations of the data sources will have the syntax to handle backquotes - there are so many non-DBMS implementations outside like elasticsearch, mongodb, etc. which I see relevant tickets in Spark JIRAs time to time.

In particular, this is a stable API. Can we update the migration guide at the very least?